.github/workflows/ci-aws.yml

```diff
@@ -0,0 +1,259 @@
+
+      - name: Build Prisma client
+        id: prisma
+        run: lerna run --scope=${{matrix.package}} prisma:generate
+
+      - name: Build code
+        id: build
+        run: lerna run --scope=${{matrix.package}} --loglevel=silent build --include-dependencies
+
+      - name: Format code
+        id: format
+        run: lerna run --scope=${{matrix.package}} --loglevel=silent check-format
+
+      - name: Lint code
+        id: lint
+        run: lerna run --scope=${{matrix.package}} --loglevel=silent lint
+
+      - name: Run tests
+        id: test
+        run: lerna run --scope=${{matrix.package}} --loglevel=silent test
+
+      - name: Discord notification
+        if: failure()
+        uses: Ilshidur/action-discord@0.3.2
+        env:
+          DISCORD_WEBHOOK: ${{ secrets.DISCORD_BUILD_NOTIFICATIONS_CHANNEL_WEBHOOK }}
+          DISCORD_EMBEDS: '[{"color":15734528,"title":"Push Validation Failed - ${{ github.ref_name }}","url":"https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}","description":"Job ${{ github.job }} in workflow ${{ github.workflow }} failed!","fields":[{"name":"Package","value":"${{ matrix.package }}"},{"name":"Commiter","value":"${{ github.event.pusher.name }}"},{"name":"Commit","value":"${{ github.event.head_commit.id }}"},{"name":"Install","value":"${{ steps.install.outcome }}","inline":true},{"name":"Build","value":"${{ steps.build.outcome }}","inline":true},{"name":"Test","value":"${{ steps.test.outcome }}","inline":true},{"name":"Format","value":"${{ steps.format.outcome }}","inline":true},{"name":"Lint","value":"${{ steps.lint.outcome }}","inline":true}]}]'
+
+  docker-build:
+    name: docker-build
+    needs: [changes]
+    if: success() #&& (github.event_name == 'push')
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16.x]
+        service: ${{ fromJSON(needs.changes.outputs.services) }} 
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+            
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }} 
+          aws-region: us-east-1
+          
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Set ecr repository
+        run: |
+          sh .github/workflows/scripts/get_ecr_repository.sh
+        env:
+          SERVICE_NAME: ${{ matrix.service }}
+      - name: Set hash
+        env:
+          working-directory: packages/${{ matrix.service }}
+        run: |
+          GITHUB_HASH_MOD=$(find ./packages/${{ matrix.service }} -type f \( -exec sha1sum {} \; \) | awk '{print $1}' | sort | sha1sum)
```

Comment:
github actions has a hash function hashFiles

Reply:
as far as i understand hashFiles is for files, we need to hash the entire folder